File: apple/HabitKit/Sources/HabitCore/Models/Weekday.swift
/// ISO-8601-Wochentag: Montag ist 1, Sonntag ist 7.
public enum Weekday: Int, Codable, CaseIterable, Sendable, Comparable, Hashable {
    case monday = 1, tuesday, wednesday, thursday, friday, saturday, sunday

    public static func < (lhs: Weekday, rhs: Weekday) -> Bool {
        lhs.rawValue < rhs.rawValue
    }

    public var isWeekend: Bool { self == .saturday || self == .sunday }
}

/// Grober Tagesabschnitt. Sortiert die Heute-Ansicht und ist später die
/// Grundlage dafür, wann ein Reminder feuert.
public enum TimeOfDay: String, Codable, CaseIterable, Sendable, Comparable {
    case morning, afternoon, evening, night

    public static func < (lhs: TimeOfDay, rhs: TimeOfDay) -> Bool {
        lhs.sortIndex < rhs.sortIndex
    }

    private var sortIndex: Int {
        switch self {
        case .morning: 0
        case .afternoon: 1
        case .evening: 2
        case .night: 3
        }
    }
}

File: server/src/domain/weekday.ts
/// Portiert aus `apple/HabitKit/Sources/HabitCore/Models/Weekday.swift`.
///
/// Der Typ `Weekday` selbst steht in `calendar.ts` — dort wird er gerechnet,
/// und `calendar.ts` soll von nichts abhängen. Hier steht, was ihn *deutet*.

import type { Weekday } from "./calendar.ts";
export type { Weekday };

export const WOCHENTAGE: readonly Weekday[] = [1, 2, 3, 4, 5, 6, 7];

export function isWeekend(tag: Weekday): boolean {
  return tag === 6 || tag === 7;
}

/// Grober Tagesabschnitt. Sortiert die Heute-Ansicht und ist später die
/// Grundlage dafür, wann ein Reminder feuert.
export type TimeOfDay = "morning" | "afternoon" | "evening" | "night";

export const TAGESZEITEN: readonly TimeOfDay[] = ["morning", "afternoon", "evening", "night"];

/// Position in `TAGESZEITEN` — der Ersatz für Swifts `Comparable`.
export function timeOfDayOrder(zeit: TimeOfDay): number {
  return TAGESZEITEN.indexOf(zeit);
}
